core/trino-main/src/main/java/io/trino/cost/AggregationStatsRule.java

```diff
@@ -50,36 +51,47 @@ public Pattern<AggregationNode> getPattern()
     @Override
     protected Optional<PlanNodeStatsEstimate> doCalculate(AggregationNode node, StatsProvider statsProvider, Lookup lookup, Session session, TypeProvider types, TableStatsProvider tableStatsProvider)
     {
-        if (node.getGroupingSetCount() != 1) {
+        if (node.getGroupingSetCount() != 1 || node.getStep() == INTERMEDIATE) {
             return Optional.empty();
         }
 
-        if (node.getStep() != SINGLE && node.getStep() != FINAL) {
-            return Optional.empty();
+        PlanNodeStatsEstimate estimate;
+
+        if (node.getStep() == PARTIAL) {
+            estimate = getPartialEstimate(statsProvider.getStats(node.getSource()),
+                    node.getGroupingKeys(),
+                    node.getAggregations());
+        }
+        else {
+            estimate = getEstimate(
+                    statsProvider.getStats(node.getSource()),
+                    node.getGroupingKeys(),
+                    node.getAggregations());
+        }
+        return Optional.of(estimate);
+    }
+
+    public static PlanNodeStatsEstimate getPartialEstimate(PlanNodeStatsEstimate sourceStats, Collection<Symbol> groupBySymbols, Map<Symbol, Aggregation> aggregations)
+    {
+        PlanNodeStatsEstimate.Builder result = PlanNodeStatsEstimate.builder();
+        // Pessimistic assumption of no reduction from PARTIAL aggregation, forwarding of the source statistics.
```

Comment:
Please add in the code comment here that this estimate makes it easier to understand CBO estimates in EXPLAIN plan output even though currently partial aggregations are added after the CBO rules have already been run.